lib/evaluate/descender.h

```diff
@@ -308,9 +308,10 @@ template<typename VISITOR> class Descender {
     auto lx{ss.lower()};
     Visit(lx);
     ss.set_lower(std::move(lx));
-    auto ux{ss.upper()};
-    Visit(ux);
-    ss.set_lower(std::move(ux));
+    if (auto ux{ss.upper()}) {
+      Visit(ux);
+      ss.set_upper(std::move(*ux));
```

Comment:
Could you briefly explain this change?

Reply:
The member function `upper` used to return an expression.  Now it returns an `optional` expression.  This change ensures that `set_upper` is called correctly with an expression.

Reply:
Okay. Was the call to `set_lower` used to be a typo?

Reply:
Yes!  I'm glad I had to make a pass through this code, it gave me a chance to notice that.